### PR TITLE
[quant][be] Clean up QAT tests in test_quantize_pt2e.py

### PR DESCRIPTION
**Pull request by andrewor14**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #108061
* __->__ #107991

Summary: This commit does 4 main things:

1. When verifying QAT numerics, automatically check both the
per tensor and the per channel cases, and automatically verify
convert numerics

2. When verifying the QAT graph, automatically check both the
per tensor and the per channel cases

3. Merge verify graph and verify numerics tests for conv-bn

4. Fix `test_prepare_qat_conv_bn_fusion_getitem_placeholder`,
which was no longer testing the right thing recent capture
changes, since the maxpool op is no longer followed by a
getitem node. However, we do still need this test for other
ops that *are* followed by getitem nodes (e.g. standalone BN).

Items (1) - (3) make the QAT tests significantly less verbose
and easier to read.

Test Plan:
python test/test_quantization.py TestQuantizePT2E
python test/test_quantization.py TestQuantizePT2EModels

Reviewers: jerryzh168, kimishpatel

Subscribers: jerryzh168, kimishpatel, supriyar

### CODE REVIEW

File: test/quantization/pt2e/test_quantize_pt2e.py [hunk 1]
```diff
@@ -1928,57 +1936,25 @@ def forward(self, x):
...
-        self._verify_symmetric_qnnpack_qat_numerics(
-            M2(), example_inputs, is_per_channel=False, verify_convert=True,
-        )
-        self._verify_symmetric_qnnpack_qat_numerics(
-            M2(), example_inputs, is_per_channel=True, verify_convert=True,
+        self._verify_symmetric_xnnpack_qat_graph(
+            m1, example_inputs, has_relu=False, has_bias=False,
         )
+        self._verify_symmetric_xnnpack_qat_numerics(m1, example_inputs)
+        self._verify_symmetric_xnnpack_qat_numerics(M2(), example_inputs)
```
Comment (jerryzh168) [inline test/quantization/pt2e/test_quantize_pt2e.py:1943]: should this be in a different test?

File: test/quantization/pt2e/test_quantize_pt2e.py [hunk 2]
```diff
@@ -2109,12 +2061,7 @@ def forward(self, x):
                 return x
         m = M()
         example_inputs = (torch.randn(1, 3, 5, 5),)
-        self._verify_symmetric_qnnpack_qat_numerics(
-            M(), example_inputs, is_per_channel=False, verify_convert=True,
-        )
-        self._verify_symmetric_qnnpack_qat_numerics(
-            M(), example_inputs, is_per_channel=True, verify_convert=True,
-        )
+        self._verify_symmetric_xnnpack_qat_numerics(M(), example_inputs)
```
Comment (jerryzh168) [inline test/quantization/pt2e/test_quantize_pt2e.py:2064]: nit: M() is instantiated in 2062, maybe remove one of them

**Review by jerryzh168 — APPROVED** [review]
lg, thanks!

**Review by jerryzh168 — COMMENTED** [review]
also might be helpful to move qat tests to a separate TestCase I think

File: test/quantization/pt2e/test_quantize_pt2e.py (hunk 1 above)
Comment (andrewor14) [reply test/quantization/pt2e/test_quantize_pt2e.py:1943]: Maybe, but I'm not sure if I want to create a test just for this case so maybe I'll just leave this here for now

**Comment by andrewor14** [comment]
> also might be helpful to move qat tests to a separate TestCase I think

Actually I think I'll do this in a separate PR so the code diff is still human legible

**Pull request closed without merging**